internal/generator/ecdsa/template/ecdsa.go.tmpl

```diff
@@ -295,7 +299,9 @@ func (privKey *PrivateKey) SignForRecover(message []byte, hFunc hash.Hash) (v ui
 		s.Add(m, s).
 			Mul(kInv, s).
 			Mod(s, order) // order != 0
-		if s.Sign() != 0 {
+
+		// ensure that s < r/2 to prevent malleability
```

Comment:
comment says s < r/2, but we check s <= r/2 -- need to fix one of the 2 :)

Reply:
ha but I guess since bHalfR is half of a prime number with a shift, it's already mixed in